cloudpathlib/cloudpath_info.py

```diff
@@ -0,0 +1,36 @@
+from functools import lru_cache
+import sys
+from typing import TYPE_CHECKING
+
+if sys.version_info < (3, 14):
+    from pathlib_abc import PathInfo
+else:
+    from pathlib.types import PathInfo
+
+if TYPE_CHECKING:
+    from .cloudpath import CloudPath
+
+
+class CloudPathInfo(PathInfo):
```

Comment:
Wondering if it's necessary to inherit from `PathInfo` here. It looks like it's just a protocol class with otherwise no implementation on it. Then we don't need to depend on pathlib-abc for now and this is just done with duck typing/structural subtyping.